sympy/physics/mechanics/functions.py

```diff
@@ -709,21 +709,13 @@ def _validate_coordinates(coordinates=None, speeds=None, check_duplicates=True,
             msgs.append(f'The auxiliary speeds {overlap_aux} are also defined '
                         f'as generalized coordinates or generalized speeds.')
     if is_dynamicsymbols:  # Check whether all coordinates are dynamicsymbols
-        for coordinate in coordinates:
-            if not (isinstance(coordinate, (AppliedUndef, Derivative)) and
-                    coordinate.free_symbols == t_set):
-                msgs.append(f'Generalized coordinate "{coordinate}" is not a '
-                            f'dynamicsymbol.')
-        for speed in speeds:
-            if not (isinstance(speed, (AppliedUndef, Derivative)) and
-                    speed.free_symbols == t_set):
-                msgs.append(
-                    f'Generalized speed "{speed}" is not a dynamicsymbol.')
-        for aux in u_auxiliary:
-            if not (isinstance(aux, (AppliedUndef, Derivative)) and
-                    aux.free_symbols == t_set):
-                msgs.append(
-                    f'Auxiliary speed "{aux}" is not a dynamicsymbol.')
+        msgs.extend(f'Generalized coordinate "{coordinate}" is not a '
+                            f'dynamicsymbol.' for coordinate in coordinates if not (isinstance(coordinate, (AppliedUndef, Derivative)) and
+                    coordinate.free_symbols == t_set))
+        msgs.extend(f'Generalized speed "{speed}" is not a dynamicsymbol.' for speed in speeds if not (isinstance(speed, (AppliedUndef, Derivative)) and
+                    speed.free_symbols == t_set))
+        msgs.extend(f'Auxiliary speed "{aux}" is not a dynamicsymbol.' for aux in u_auxiliary if not (isinstance(aux, (AppliedUndef, Derivative)) and
+                    aux.free_symbols == t_set))
```

Comment:
This is not preferred code nor does it abide by our line length guideline. Please revert. In fact, please revert all changes in the physics/mechanics/biomechanics/continuum_mechanics packages. Blindly applying such "optimizations" is not generally helpful. Authors write the code then way they want to and that may be based on readability versus nano-performance.

If you want to work on performance optimizations, my suggestion would be to start with profiling SymPy and creating benchmarks. If we do have code that significantly slows SymPy's performance, changes for performance that neglect readability can be made.

Reply:
> Do we have any lints for the linelength guidelines?

No, SymPy doesn't have many lints because we have traditionally been fine with things being a guideline not a hard rule. For line length, there are times when it is fine to let the line be longer than 79 characters. But in general, we keep them under 80 characters.